### PR TITLE
[ticket/14437] Correctly assume index from attachment display order

### PR DESCRIPTION
**Pull request by marc1706**
Ticket: https://tracker.phpbb.com/browse/PHPBB3-14437

### CODE REVIEW

File: phpBB/includes/functions_content.php [hunk 1]
```diff
@@ -1248,6 +1236,18 @@ function parse_attachments($forum_id, &$message, &$attachments, &$update_count,
 
 	$unset_tpl = array_unique($unset_tpl);
 
+	// Sort correctly
+	if ($config['display_order'])
+	{
+		// Ascending sort
```
Comment (rxu) [inline phpBB/includes/functions_content.php]: Descending sort

File: phpBB/includes/functions_content.php [hunk 2]
```diff
@@ -1248,6 +1236,18 @@ function parse_attachments($forum_id, &$message, &$attachments, &$update_count,
...
+		krsort($attachments);
+	}
+	else
+	{
+		// Descending sort
+		ksort($attachments);
```
Comment (rxu) [inline phpBB/includes/functions_content.php]: Ascending sort

File: phpBB/includes/functions_content.php (hunk 1 above)
Comment (marc1706) [inline phpBB/includes/functions_content.php]: This is the same as under the acp settings for the display order. Why do you think it should be the other way round?

Comment (rxu) [inline phpBB/includes/functions_content.php]: Dunno, but isn't it what ```krsort``` does?

Comment (rxu) [inline phpBB/includes/functions_content.php]: ok, probably despite it's descending technically, it's acsending in respect of the attachments order.

Comment (marc1706) [inline phpBB/includes/functions_content.php]: Yeah, the idea behind this might probably be somewhat confusing. ;)

**Pull request merged**